$OpenBSD: patch-shell_ev-window_c,v 1.2 2009/08/17 09:57:44 ajacoutot Exp $

Taken from http://bugzilla.gnome.org/show_bug.cgi?id=571051

This prevents evince from stealing focus when reloading a document.
Evince reloads documents automatically when they are changed while
being displayed. Very annoying while editing LaTeX files and rebuilding
the corresponding PDF.

--- shell/ev-window.c.orig	Mon May 18 22:23:19 2009
+++ shell/ev-window.c	Sun Aug 16 13:39:38 2009
@@ -1027,6 +1027,11 @@ setup_size_from_metadata (EvWindow *window)
 	GValue x = { 0, };
 	GValue y = { 0, };
 
+	/* Don't try to change the window when reloading a document */
+	if (window->priv->in_reload) {
+	    return;
+	}
+
 	if (ev_metadata_manager_get (uri, "window_maximized", &maximized, FALSE)) {
 		if (g_value_get_boolean (&maximized)) {
 			gtk_window_maximize (GTK_WINDOW (window));
